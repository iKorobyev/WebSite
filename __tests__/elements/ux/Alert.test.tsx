import React from 'react'
import { cleanup, render, waitFor, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Alert, AlertVariantType } from '~ux'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Alert', () => {
  it('renders correctly', async () => {
    const props = {
      message: 'Test',
      close: () => null,
      options: { type: 'success' as AlertVariantType }
    }

    const { container } = render(<Alert {...props} />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
