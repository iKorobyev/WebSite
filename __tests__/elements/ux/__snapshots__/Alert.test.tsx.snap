// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div>
  <div
    class="w-fit min-h-40 flex font-medium items-center justify-center rounded-5 px-30 bg-green-700"
    data-testid="alert"
  >
    Test
  </div>
</div>
`;
